csv/src/fs2/data/csv/RowDecoder.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2019 Lucas Satabin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package fs2.data.csv
+
+import cats._
+import cats.data.NonEmptyList
+
+/** Describes how a row can be decoded to the given type.
+  */
+trait RowDecoder[T] {
+  def apply(cells: NonEmptyList[String]): DecoderResult[T]
+}
+
+object RowDecoder {
+
+  implicit object RowDecoderFunctor extends Functor[RowDecoder] {
+    def map[A, B](fa: RowDecoder[A])(f: A => B): RowDecoder[B] =
+      cells => fa(cells).map(f)
+  }
+
+  def apply[T: RowDecoder]: RowDecoder[T] = implicitly[RowDecoder[T]]
+
```

Comment:
Maybe we could add a conversion utility from `CsvRowDecoder` to `RowDecoder` here?

Reply:
yep, I thought about it but forgot, I will add it as well.

Reply:
Actually, it is only possible to convert in the other direction, as a `CsvRowDecoder` might need headers that a `RowDecoder` doesn't have, I will add the other conversion.